### PR TITLE
chore(cloud-auth-nonce): update CloudAuthSIWx to provide the nonce header

### PR DESCRIPTION
**Pull request by Cali93**
# Description

We've updated the Cloud Auth API to send a different header when making the authenticate request.

## Type of change

- [x] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [x] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

**Review by zoruka — COMMENTED** [review]
We need to update and add tests for the changes

File: .changeset/metal-coins-guess.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+'@reown/appkit-common': patch
+'@reown/appkit-experimental': patch
+'@reown/appkit-polyfills': patch
+'@reown/appkit-siwe': patch
+'@reown/appkit-ui': patch
+'@reown/appkit-wallet': patch
+'@reown/appkit-wallet-button': patch
+---
+
+Fix logic for authentication header on CloudAuthSIWX"
```
Comment (magiziz) [inline .changeset/metal-coins-guess.md:22]: Remove `"` at the end ?

File: packages/siwx/src/configs/CloudAuthSIWX.ts [hunk 2]
```diff
@@ -18,11 +18,13 @@ import { InformalMessenger } from '../index.js'
...
  */
 export class CloudAuthSIWX implements SIWXConfig {
-  private readonly localStorageKey: string
+  private readonly localAuthStorageKey: string
+  private readonly localNonceStorageKey: string
   private readonly messenger: SIWXMessenger
 
   constructor(params: CloudAuthSIWX.ConstructorParams = {}) {
-    this.localStorageKey = params.localStorageKey || '@appkit/siwx-token'
+    this.localAuthStorageKey = params.localAuthStorageKey || '@appkit/siwx-auth-token'
```
Comment (enesozturk) [inline packages/siwx/src/configs/CloudAuthSIWX.ts:26]: Can we use these keys from `SafeLocalStorage` util

**Review by magiziz — APPROVED** [review]

File: packages/siwx/tests/configs/CloudAuthSIWX.test.ts [hunk 3]
```diff
@@ -137,12 +137,12 @@ Issued At: 2024-12-05T16:02:32.905Z`)
...
           body: '{"message":"Hello AppKit!","signature":"0x3c70e0a2d87f677dc0c3faf98fdf6313e99a3d9191bb79f7ecfce0c2cf46b7b33fd4c4bb83bca82fe872e35963382027d0d18018342d7dc36a675918cb73e9061c","clientId":null}',
           headers: {
-            Authorization: 'Bearer mock_nonce_token'
+            'x-nonce-jwt': 'Bearer mock_nonce_token'
           },
           method: 'POST'
         }
       )
-      expect(setItemSpy).toHaveBeenCalledWith('@appkit/siwx-token', 'mock_authenticate_token')
+      expect(setItemSpy).toHaveBeenCalledWith('@appkit/siwx-auth-token', 'mock_authenticate_token')
```
Comment (tomiir) [inline packages/siwx/tests/configs/CloudAuthSIWX.test.ts:145]: don't we need a test for the `Authorization` case? same for other places where this was changed

File: packages/siwx/src/configs/CloudAuthSIWX.ts (hunk 2 above)
Comment (tomiir) [reply packages/siwx/src/configs/CloudAuthSIWX.ts:26]: I think this should be independant from SafeLocalStorage. 

But would be nice to use the key from its constants

File: packages/siwx/src/configs/CloudAuthSIWX.ts [hunk 4]
```diff
@@ -114,22 +130,23 @@ export class CloudAuthSIWX implements SIWXConfig {
...
-  private setStorageToken(token: string): void {
-    localStorage.setItem(this.localStorageKey, token)
+  private setStorageToken(token: string, key: string): void {
+    localStorage.setItem(key, token)
   }
 
-  private clearStorageToken(): void {
-    localStorage.removeItem(this.localStorageKey)
+  private clearStorageTokens(): void {
+    localStorage.removeItem(this.localAuthStorageKey)
```
Comment (enesozturk) [inline packages/siwx/src/configs/CloudAuthSIWX.ts:142]: We should use `SafeLocalStorage` utility - it's checking window object for possible SSR issues. Let's switch to it

File: packages/siwx/src/configs/CloudAuthSIWX.ts [hunk 5]
```diff
@@ -114,22 +130,23 @@ export class CloudAuthSIWX implements SIWXConfig {
...
+    localStorage.removeItem(this.localNonceStorageKey)
```
[the remark this answers is not in the document]
Comment (zoruka) [reply packages/siwx/src/configs/CloudAuthSIWX.ts:143]: the keys are overridable, but we could remove it if you guys prefer

**Review by tomiir — APPROVED** [review]

**Pull request merged**